src/crystal/main.cr

```diff
@@ -56,7 +56,9 @@ module Crystal
         1
       end
 
-    AtExitHandlers.run status
+    new_status = AtExitHandlers.run status
+    status = {new_status, status}.max
```

Comment:
last status should win, dont think max makes sense. ruby not work this way.

Reply:
Yeah I'm not sure about this too, was just a try anyway!
My idea was: if the handler has an `exit` (or `exit 0`) and the process is exiting with a `exit 1` (or more), then it should not exit with a status code `0`.

No idea how Ruby works about that.

Reply:
ruby seem to use status code from exit that was last executed. each exit changes status code

Reply:
Indeed, please use the latest exist code.